server/paymentProviders/stripe/payment-intent.ts

```diff
@@ -32,21 +32,51 @@ async function processNewOrder(order: typeof models.Order) {
     description: order.description,
   };
 
-  if (applicationFee && isPlatformRevenueDirectlyCollected && hostStripeAccount.username !== config.stripe.accountId) {
+  const isPlatformHost = hostStripeAccount.username === config.stripe.accountId;
+  const isSavePaymentMethod = order.data?.savePaymentMethod || order.interval;
+
+  if (applicationFee && isPlatformRevenueDirectlyCollected && !isPlatformHost) {
     // eslint-disable-next-line camelcase
     paymentIntentParams.application_fee_amount = convertToStripeAmount(order.currency, applicationFee);
   }
 
-  if (order.data?.savePaymentMethod) {
+  if (isSavePaymentMethod) {
     // eslint-disable-next-line camelcase
     paymentIntentParams.setup_future_usage = 'off_session';
   }
 
+  let stripeCustomerAccount = await order.fromCollective.getCustomerStripeAccount(hostStripeAccount.username);
+  if (isSavePaymentMethod && !stripeCustomerAccount) {
+
+    const customer = await stripe.customers.create(
+      {
+        email: order.createdByUser.email,
+        description: `${config.host.website}/${order.fromCollective.slug}`,
+      },
+      !isPlatformHost
+        ? {
+            stripeAccount: hostStripeAccount.username,
+          }
+        : undefined,
+    );
+
+    stripeCustomerAccount = await models.ConnectedAccount.create({
+      clientId: hostStripeAccount.username,
+      username: customer.id,
+      CollectiveId: order.fromCollective.id,
+      service: Service.STRIPE_CUSTOMER,
+    });
+
+    paymentIntentParams.customer = customer.id;
+  } else if (stripeCustomerAccount) {
+    paymentIntentParams.customer = stripeCustomerAccount.username;
+  }
+
   try {
     const paymentIntent = await stripe.paymentIntents.update(order.data.paymentIntent.id, paymentIntentParams, {
       stripeAccount: hostStripeAccount.username,
     });
-    await order.update({ status: OrderStatuses.PROCESSING, data: { ...order.data, paymentIntent } });
```

Comment:
@kewitz  Removing this status update here to keep the order as NEW, as the payment intent was not yet confirmed (might not be confirmed) and can be misleading to show this order as processing.